crates/oxc_semantic/src/scope.rs

```diff
@@ -184,16 +184,20 @@ impl ScopeTree {
         let scope_id = self.parent_ids.push(parent_id);
         _ = self.flags.push(flags);
         _ = self.bindings.push(Bindings::default());
+        // XXX: should this be a real invalid sentinel value or an Option?
+        _ = self.node_ids.push(AstNodeId::dummy());
 
-        if let Some(parent_id) = parent_id {
-            self.child_ids.entry(parent_id).or_default().push(scope_id);
-        }
+        let parent = match parent_id {
+            Some(id) => vec![id],
+            None => vec![],
+        };
+        _ = self.child_ids.push(parent);
```

Comment:
I don't think this is correct. Unless I've misunderstood, it's setting a scope's parent as it's own child. I think these 5 lines should be:

```rs
// Set this scope as child of parent scope
if let Some(parent_id) = parent_id {
    self.child_ids[parent_id].push(scope_id);
}
// Init list of child scopes for this scope (currently empty)
_ = self.child_ids.push(vec![]);
```